src/etc/inc/util.inc

```diff
@@ -1932,19 +1932,34 @@ function mac_format($clientmac) {
 	}
 }
 
-function resolve_retry($hostname, $retries = 5) {
-
-	if (is_ipaddr($hostname)) {
-		return $hostname;
-	}
+function resolve_retry($hostname, $protocol = 'inet', $retries = 5) {
 
 	for ($i = 0; $i < $retries; $i++) {
-		// FIXME: gethostbyname does not work for AAAA hostnames, boo, hiss
-		$ip = gethostbyname($hostname);
-
-		if ($ip && $ip != $hostname) {
-			/* success */
-			return $ip;
+		switch ($protocol) {
+			case 'inet': 
+				$checkproto = 'is_ipaddrv4';
+				$ipproto = 'ip';
+				$dnsproto = array('ip' => DNS_A);
+				break;
+			case 'inet6':
+				$checkproto = 'is_ipaddrv6';
+				$dnsproto = array('ipv6' => DNS_AAAA);
+				break;
+			case 'any':
+				$checkproto = 'is_ipaddr';
+				$dnsproto = array('ip' => DNS_A, 'ipv6' => DNS_AAAA);
+				break;
+			default:
+				break;
+		}
+		if (call_user_func($checkproto, $hostname)) {
+			return $hostname;
+		}
+		foreach ($dnsproto as $ipp => $dnsp) {
+			$ip = dns_get_record($hostname, $dnsp)[0][$ipp];
```

Comment:
I know you said you still received an error, but this is still the wrong way to do it. It must get the result and check it before treating it as an array. You could suppress any remaining errors by using `$result = @dns_get_record( [...]` but since that function may not return an array, calling it this way is incorrect.

Reply:
done